Makefile

```diff
@@ -0,0 +1,60 @@
+.DEFAULT_GOAL=all
+.PHONY: clean all copy_env_files lemur_checkout build_containers restart_containers inspect_image_size
+
+# Shell to use for running scripts
+SHELL := $(shell which bash)
+# Get docker path or an empty string
+DOCKER := $(shell command -v docker)
+# Get docker-compose path or an empty string
+DOCKER_COMPOSE := $(shell command -v docker-compose)
+
+# Test if the dependencies we need to run this Makefile are installed
+deps:
+ifndef DOCKER
+	@echo "Docker is not available. Please install docker"
+	@exit 1
+endif
+ifndef DOCKER_COMPOSE
+	@echo "docker-compose is not available. Please install docker-compose"
+	@exit 1
+endif
+
+# TODO: for future use, will tag docker build
+COMMIT := $(shell git rev-list -1 HEAD)
+BRANCH := $(shell git rev-parse --abbrev-ref HEAD)
+VERSION := $(shell git describe --tags --always)
+
+LEMUR_GIT_DIR = lemur-build-docker/.lemur
+
+clean: deps
+	rm -rf $(LEMUR_GIT_DIR)
+	docker-compose down -v || sudo docker-compose down -v
+
+all: deps
+	$(MAKE) copy_env_files
+	$(MAKE) lemur_checkout
+	$(MAKE) build_containers
+	$(MAKE) restart_containers
+
+copy_env_files:
+	[ -f .lemur.env ] || cp .lemur.env.dist .lemur.env && echo ".lemur.env created from scratch"
+	[ -f .pgsql.env ] || cp .pgsql.env.dist .pgsql.env && echo ".pgsql.env created from scratch"
+
+lemur_checkout:
+	[ -d $(LEMUR_GIT_DIR) ] || git clone --depth=1 https://github.com/Netflix/lemur.git $(LEMUR_GIT_DIR)
+	cd $(LEMUR_GIT_DIR) && git pull
+
+build_containers: deps
+	docker-compose build || sudo docker-compose build || echo "failed to build containers"
```

Comment:
Do you know of a case where `sudo` is necessary? The `build` command failed for a non-permissions related reason for me, and I know I don't need `sudo` so I'm hesitant to prompt the user for it for no reason (and in general want to avoid `sudo` if we don't need it).

Reply:
I don't know how it can be fixed. Sudo right need if user working on linux server and do not have permission to run docker. I'v added TODO for this.

An example:
![Screenshot 2021-02-04 at 08 02 46](https://user-images.githubusercontent.com/64213648/106847159-6a1cef80-66bf-11eb-9b20-5d49c23bd0b6.png)

Reply:
We have 2 choice here:
- completely removing sudo from all commands. But in this case all commands will run with sudo. E.g. user run this make file with sudo because docker.
- will make/find more elegant way to detect rights to the docker

Reply:
I suppose I didn't run into this as I followed [these post-installation steps](https://docs.docker.com/engine/install/linux-postinstall/#manage-docker-as-a-non-root-user), but I guess that if you don't, you'd need `sudo`.

We can leave this for now and rethink it later.